server/socketEvents.js

```diff
@@ -0,0 +1,26 @@
+exports = module.exports = function (io) {
+    // Set socket.io listeners.
+    io.on('connection', (socket) => {
+        console.log('a user connected: ' + socket.id);
+
+        socket.on('new message', (msgObject) => {
+
+            const reply = new Message({
+                conversationId: msgObject.conversationId,
+                body: msgObject.msg,
+                author: msgObject.userid
+            });
+
+            reply.save((err, sentReply) => {
+                if (err) {
+                    return next(err);
+                }
+            });
+            io.sockets.emit('refresh message', msgObject.conversationId);
```

Comment:
I think the server should send the array of messages when you emit 'refresh message', otherwise we'll have to make an API request in the client each time